fedora_messaging/tests/unit/test_config.py

```diff
@@ -410,3 +410,15 @@ def test_load_on_setup_logging(self, mock_exists, mock_log):
         mock_log.info.assert_called_once_with(
             "Loading configuration from /etc/fedora-messaging/config.toml"
         )
+
+    def test_load_on_get_item(self):
+        """Assert load_config is called when __getitem__ is invoked."""
+        config = msg_config.LazyConfig()
+        config.load_config = mock.Mock()
+
+        try:
+            config["some_key"]
+        except KeyError:
+            pass
+
+        config.load_config.assert_called_once_with()
```

Comment:
This particular test was because I saw one line was being uncovered, the line where load_config is called on first access, so I figured I'd add a test for it for 100% coverage. 

I could add a separate test to assert the validate_* functions are getting called, but they're not hidden behind any if logic and there are tests to make sure they validate a complete config so I think it might be a bit zealous.

Reply:
Okay :-)